### PR TITLE
Handle vm_instance_view.statuses being None

### LINKED ISSUE #2569: Resource processor error on deploying user-resource: `TypeError: 'NoneType' object is not iterable`
**Describe the bug**
Whilst attempting to deploy a user resource (Linux VM), I encountered a failure. Inspecting the resource processor logs revealed the following error:

```
Exception in ASGI application
Traceback (most recent call last):
File "/usr/local/lib/python3.8/site-packages/anyio/streams/memory.py", line 94, in receive
return self.receive_nowait()
File "/usr/local/lib/python3.8/site-packages/anyio/streams/memory.py", line 89, in receive_nowait
raise WouldBlock
anyio.WouldBlock

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 41, in call_next
message = await recv_stream.receive()
File "/usr/local/lib/python3.8/site-packages/anyio/streams/memory.py", line 114, in receive
raise EndOfStream
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 375, in run_asgi
result = await app(self.scope, self.receive, self.send)
File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
return await self.app(scope, receive, send)
File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 261, in __call__
await super().__call__(scope, receive, send)
File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 112, in __call__
await self.middleware_stack(scope, receive, send)
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
raise exc
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
await self.app(scope, receive, _send)
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
raise exc
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
await self.app(scope, receive, _send)
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 63, in __call__
response = await self.dispatch_func(request, call_next)
File "/api/./services/tracing.py", line 83, in dispatch
response = await call_next(request)
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 44, in call_next
raise app_exc
File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 34, in coro
await self.app(scope, request.receive, send_stream.send)
File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
raise exc
File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
await self.app(scope, receive, sender)
File "/usr/local/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
raise e
File "/usr/local/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
await self.app(scope, receive, send)
File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 656, in __call__
await route.handle(scope, receive, send)
File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 259, in handle
await self.app(scope, receive, send)
File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 61, in app
response = await func(request)
File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 227, in app
raw_response = await run_endpoint_function(
File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 160, in run_endpoint_function
return await dependant.call(**values)
File "/api/./api/routes/workspaces.py", line 293, in retrieve_user_resources_for_workspace_service
user_resource.azureStatus = get_azure_resource_status(user_resource.properties['azure_resource_id'])
File "/api/./services/azure_resource_status.py", line 17, in get_azure_resource_status
power_state = [x for x in vm_instance_view.statuses if x.code.startswith('PowerState')][0].display_status
TypeError: 'NoneType' object is not iterable
```

### PR DESCRIPTION
**Pull request by stuartleeks**
# Resolves #2569

## What is being addressed

In some situations, it seems that `vm_instance_view.statuses` is `None` and this caused an exception

## How is this addressed

- update API code to handle `vm_isntance_view.statuses` being `None` when iterating to find VM status

### CODE REVIEW

File: api_app/services/azure_resource_status.py [hunk 1]
```diff
@@ -14,7 +14,11 @@ def get_azure_resource_status(resource_id):
     try:
         if resource_type == 'Microsoft.Compute/virtualMachines':
             vm_instance_view: models.VirtualMachineInstanceView = get_azure_vm_instance_view(resource_name, resource_group_name)
-            power_state = [x for x in vm_instance_view.statuses if x.code.startswith('PowerState')][0].display_status
+            power_state = None
+            if vm_instance_view.statuses is not None:
```
Comment (martinpeck) [inline api_app/services/azure_resource_status.py:18]: I think you could have done something like...

```
[k for k in (tags or [])]
```

... but being more verbose is probably better here so ignore this.

**Review by martinpeck — APPROVED** [review]

File: api_app/services/azure_resource_status.py (hunk 1 above)
Comment (stuartleeks) [reply api_app/services/azure_resource_status.py:18]: Oooh, that's good to know. That line is already has a reasonable amount of complexity, so I agree that verbose works well here.

**Comment by tamirkamara** [comment]
@stuartleeks aside from this fix you might want to double check the base workspace you're using as this problem suggests an old version that doesn't assign the api identity vm contributor over the workspace resource group. Some @anatbal had seen just today.

**Pull request merged**